### PR TITLE
finish the translation of package container and sort

### PR DESCRIPTION
**Pull request by bqdong**
This PR will be imported into Gerrit with the title and first
comment (this text) used to generate the subject and body of
the Gerrit change.

**Please ensure you adhere to every item in this list.**

More info can be found at https://github.com/golang/go/wiki/CommitMessage

+ The PR title is formatted as follows: `net/http: frob the quux before blarfing`
  + The package name goes before the colon
  + The part after the colon uses the verb tense + phrase that completes the blank in,
    "This change modifies Go to ___________"
  + Lowercase verb after the colon
  + No trailing period
  + Keep the title as short as possible. ideally under 76 characters or shorter
+ No Markdown
+ The first PR comment (this one) is wrapped at 76 characters, unless it's
  really needed (ASCII art, table, or long link)
+ If there is a corresponding issue, add either `Fixes #1234` or `Updates #1234`
  (the latter if this is not a complete fix) to this comment
+ If referring to a repo other than `golang/go` you can use the
  `owner/repo#issue_number` syntax: `Fixes golang/tools#1234`
+ We do not use Signed-off-by lines in Go. Please don't add them.
  Our Gerrit server & GitHub bots enforce CLA compliance instead.
+ Delete these instructions once you have read and applied them

### CODE REVIEW

File: src/container/heap/example_pq_test.go [hunk 1]
```diff
@@ -2,29 +2,29 @@
...
 
-// A PriorityQueue implements heap.Interface and holds Items.
+// PriorityQueue 实现了 heap.Interface，并保存了 Item。
 type PriorityQueue []*Item
 
 func (pq PriorityQueue) Len() int { return len(pq) }
 
 func (pq PriorityQueue) Less(i, j int) bool {
-	// We want Pop to give us the highest, not lowest, priority so we use greater than here.
+	// 我们希望 Pop 给我们最高而非最低的优先级，因此我们使用在这里使用大于。
```
Comment (polaris1119) [inline src/container/heap/example_pq_test.go:27]: 多了个 “使用”

File: src/container/heap/heap.go [hunk 2]
```diff
@@ -2,43 +2,40 @@
...
 }
 
-// Init establishes the heap invariants required by the other routines in this package.
-// Init is idempotent with respect to the heap invariants
-// and may be called whenever the heap invariants may have been invalidated.
-// The complexity is O(n) where n = h.Len().
+
+// Init 建立此程序包中其他例程所需的堆不变式。
+// Init 对于堆不变量是等幂的，并且可以在堆不变量无效时调用。
+// n = h.Len() 处的复杂度是 O(log n)。
```
Comment (polaris1119) [inline src/container/heap/heap.go:38]: 这段有些问题。invariants 译为约束性如何？
该句的大概意思：一个堆在使用任何堆操作之前应先初始化。Init函数对于堆的约束性是幂等的（多次执行无意义），并可能在任何时候堆的约束性被破坏时被调用。

你看如何调整下

File: src/container/heap/heap.go [hunk 3]
```diff
@@ -47,25 +44,25 @@ func Init(h Interface) {
 	}
 }
 
-// Push pushes the element x onto the heap.
-// The complexity is O(log n) where n = h.Len().
+// Push 将元素 x 压入堆中。
+// n = h.Len() 处的复杂度是 O(log n)。
```
Comment (polaris1119) [inline src/container/heap/heap.go:48]: 建议改为：复杂度是 O(log n)，n = h.Len() 

下同

怎么感觉原文 大 O 是 0？

File: src/container/list/list.go [hunk 4]
```diff
@@ -2,85 +2,84 @@
...
+// Back 返回链表中最后一个元素，链表为空时返回 nil。
 func (l *List) Back() *Element {
 	if l.len == 0 {
 		return nil
 	}
 	return l.root.prev
 }
 
+// lazyInit 惰性地初始化一个零列表值。
 // lazyInit lazily initializes a zero List value.
```
Comment (polaris1119) [inline src/container/list/list.go]: 这一句英文没删除

**Review by polaris1119 — CHANGES_REQUESTED** [review]

File: src/container/heap/heap.go (hunk 2 above)
Comment (bqdong) [reply src/container/heap/heap.go:38]: 这一段有些不太会翻译，就按照你说的那个翻译，行吗？

File: src/container/list/list.go (hunk 4 above)
Comment (bqdong) [reply src/container/list/list.go]: 这个函数不出现在文档里，就可以不用翻译了，我把翻译删掉

File: src/container/heap/heap.go (hunk 3 above)
Comment (bqdong) [reply src/container/heap/heap.go:48]: 我看了一下，原文是 O ，不是 0 (零）。我把翻译调整一下

File: src/container/list/list.go [hunk 5]
```diff
@@ -2,85 +2,84 @@
...
-// Back returns the last element of list l or nil if the list is empty.
+// Back 返回链表中最后一个元素，链表为空时返回 nil。
 func (l *List) Back() *Element {
 	if l.len == 0 {
 		return nil
 	}
 	return l.root.prev
 }
 
+// lazyInit 惰性地初始化一个零列表值。
```
Comment (darcyaf) [inline src/container/list/list.go]:
```suggestion
```

**Pull request merged**